File: src/com/mycompany/myapp/entities/Compte.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author masso
 */
public class Compte {
    protected String nomDutilisateur;
    protected int id;
    protected String nom;
    protected String prenom;
    protected int age ;
    protected String adresseMail;
    protected String motDePasse;
    protected int numTel;
    protected Date date;
    protected String type;
    
    public Compte() {
        
    }

    public Compte(int id) {
        this.id = id;
    }

    public Compte(String nomDutilisateur, int id, String nom, String prenom, int age, String adresseMail, String motDePasse, int numTel, Date date, String type) {
        this.nomDutilisateur = nomDutilisateur;
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.adresseMail = adresseMail;
        this.motDePasse = motDePasse;
        this.numTel = numTel;
        this.date = date;
        this.type = type;
    }

    public Compte(String nomDutilisateur, String nom, String prenom, int age, String adresseMail, String motDePasse, int numTel, Date date, String type) {
        this.nomDutilisateur = nomDutilisateur;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.adresseMail = adresseMail;
        this.motDePasse = motDePasse;
        this.numTel = numTel;
        this.date = date;
        this.type = type;
    }
    
    public Compte(int id,String nomDutilisateur, String nom, String prenom, int age, String adresseMail, String motDePasse, int numTel) {
        this.nomDutilisateur=nomDutilisateur;
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.adresseMail = adresseMail;
        this.motDePasse = motDePasse;
        this.numTel = numTel;
    }

    public Compte(String nomDutilsiateur ,String nom, String prenom, int age, String adresseMail, String motDePasse, int numTel) {
        this.nomDutilisateur=nomDutilsiateur;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.adresseMail = adresseMail;
        this.motDePasse = motDePasse;
        this.numTel = numTel;
    }

    public String getNom() {
        return nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public int getAge() {
        return age;
    }

    public String getAdresseMail() {
        return adresseMail;
    }

    public String getMotDePasse() {
        return motDePasse;
    }

    public int getNumTel() {
        return numTel;
    }

    public int getId() {
        return id;
    }

    public String getNomDutilisateur() {
        return nomDutilisateur;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setAdresseMail(String adresseMail) {
        this.adresseMail = adresseMail;
    }

    public void setMotDePasse(String motDePasse) {
        this.motDePasse = motDePasse;
    }

    public void setNumTel(int numTel) {
        this.numTel = numTel;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setNomDutilisateur(String nomDutilisateur) {
        this.nomDutilisateur = nomDutilisateur;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Compte{" + "nomDutilisateur=" + nomDutilisateur + ", id=" + id + ", nom=" + nom + ", prenom=" + prenom + ", age=" + age + ", adresseMail=" + adresseMail + ", motDePasse=" + motDePasse + ", numTel=" + numTel + ", date=" + date + ", type=" + type + '}';
    }
    
    
    
}

File: src/Entity/Compte.php
<?php

namespace App\Entity;
use Symfony\Component\Security\Core\User\UserInterface;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Compte
 *
 * @ORM\Table(name="compte")
 * @ORM\Entity
 */
class Compte implements UserInterface
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
        * @Assert\NotNull
        * @Groups("post:read")
     * @ORM\Column(name="userName", type="string", length=32, nullable=false)
     */
    private $username;

    /**
     * @var string
     *
      * @Groups("post:read")
     * @ORM\Column(name="nom", type="string", length=32, nullable=false)
     */
    private $nom;

    /**
     * @var string
     *
     * @Groups("post:read")
     * @ORM\Column(name="prenom", type="string", length=32, nullable=false)
     */
    private $prenom;

    /**
     * @var int
     *
     * @Groups("post:read")
     * @ORM\Column(name="age", type="integer", nullable=false)
     */
    private $age;

    /**
     * @var string
     * @Groups("post:read")
     * @Assert\NotNull
     * @Assert\Email(
     *     message = "The email '{{ value }}' is not a valid email.",
     *     checkMX = true
     * )
     * @ORM\Column(name="adresse_mail", type="string", length=64, nullable=false)
     */
    private $adresseMail;

    /**
     * @var string
     * @Groups("post:read")
     * @ORM\Column(name="mot_de_passe", type="string", length=32, nullable=false)
     */
    private $motDePasse;

    /**
     * @var int
     *  @Groups("post:read")
     * @ORM\Column(name="num_tel", type="integer", nullable=false)
     */
    private $numTel;

    /**
     * @var \DateTime|null
     *  @Groups("post:read")
     * @ORM\Column(name="date", type="date", nullable=true)
     */
    private $date;

    /**
     * @Groups("post:read")
     * @ORM\Column(type="json")
     */
    private $roles = [];


    /**
     * @var string
     * @Groups("post:read")
     * @ORM\Column(name="type", type="string", length=32, nullable=false)
     */
    private $type;

    protected $captchaCode;

    public function getCaptchaCode()
    {
        return $this->captchaCode;
    }

    public function setCaptchaCode($captchaCode)
    {
        $this->captchaCode = $captchaCode;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;

        return $this;
    }

    public function getUsername(): ?string
    {
        return $this->username;
    }

    public function setUsername(string $username): self
    {
        $this->username = $username;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getAge(): ?int
    {
        return $this->age;
    }

    public function setAge(int $age): self
    {
        $this->age = $age;

        return $this;
    }

    public function getAdresseMail(): ?string
    {
        return $this->adresseMail;
    }

    public function setAdresseMail(string $adresseMail): self
    {
        $this->adresseMail = $adresseMail;

        return $this;
    }

    public function getMotDePasse(): ?string
    {
        return $this->motDePasse;
    }

    public function setMotDePasse(string $motDePasse): self
    {
        $this->motDePasse = $motDePasse;

        return $this;
    }

    public function getNumTel(): ?int
    {
        return $this->numTel;
    }

    public function setNumTel(int $numTel): self
    {
        $this->numTel = $numTel;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(?\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }




    public function getRoles(): array
    {
        $roles = $this->roles;
        // guarantee every user at least has ROLE_USER
        $roles[] = 'ROLE_USER';

        return array_unique($roles);
    }

    public function getPassword()
    {
        // TODO: Implement getPassword() method.
    }

    public function getSalt()
    {
        // TODO: Implement getSalt() method.
    }

    public function eraseCredentials()
    {
        // TODO: Implement eraseCredentials() method.
    }
}
